File: src/main/resources/static/js/service/dddetailService.js
//服务层
app.service('dddetailService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../dddetail/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../dddetail/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../dddetail/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../dddetail/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../dddetail/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../dddetail/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../dddetail/search.do?page='+page+"&rows="+rows, searchEntity);
	}
    //查询明细
    this.searchDet=function(ddno){
        return $http.post('../dddetail/searchDet.do?ddno='+ddno);
    }
});

File: src/main/java/com/zzti/fruits/service/DddetailService.java
package com.zzti.fruits.service;
import java.util.List;

import com.zzti.fruits.entity.PageResult;
import com.zzti.fruits.pojo.Dddetail;
import com.zzti.fruits.pojogroup.GroupDdDet;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface DddetailService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<Dddetail> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(Dddetail dddetail);
	
	
	/**
	 * 修改
	 */
	public void update(Dddetail dddetail);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public Dddetail findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(Dddetail dddetail, int pageNum, int pageSize);

	List<GroupDdDet> searchDet(String ddno);
}
